lib/internal/Magento/Framework/View/Result/Layout.php

```diff
@@ -7,6 +7,7 @@
 namespace Magento\Framework\View\Result;
 
 use Magento\Framework;
+use Magento\Framework\App\Response\HttpInterface as HttpResponseInterface;
```

Comment:
From what I understood `\Magento\Framework\Controller\ResultInterface` should be HTTP agnostic. I guess there could be CLI result types for example.
If it doesn't have to be, then all the methods of `HttpInterface` could go straight into `ResultInterface`, and no new interface needed to be introduced.